spec/abstract-jobs.html

```diff
@@ -44,38 +44,131 @@ <h1> Abstract Jobs </h1>
             <td><ins>List of objects</ins></td>
             <td><ins>Initially a new empty List, representing the list of objects to be kept alive until the end of the current Job</ins></td>
           </tr>
-          <tr>
-            <td><ins>[[FinalizationGroups]]</ins></td>
-            <td><ins>List of FinalizationGroup objects</ins></td>
-            <td><ins>Initially a new empty List, representing the set of all FinalizationGroups</ins></td>
-          </tr>
         </tbody>
       </table>
     </emu-table>
 
-  <p>
-    At any given time, if all references to a ECMAScript object are from a
-    [[Target]] field or internal slot, and KeepDuringJob was not called on the
-    object during current RunJobs algorithm/since the last microtask checkpoint,
-    the implementation MAY replace the reference to the object in all such
-    [[Target]] fields or internal slots with the value ~empty~.
-  </p>
+  <emu-clause id="sec-weakref-processing-model">
+    <h1>Processing model of WeakRefs and FinalizationGroups</h1>
 
-  <emu-clause id="sec-do-agent-finalization" aoid="DoAgentFinalization">
-    <h1> DoAgentFinalization ( )</h1>
-    <p> The following steps are performed: </p>
+    <emu-clause id="sec-weakref-invariants">
+      <h1>Objectives</h1>
+
+      <p>
+        This specification does not make any guarantees about garbage collection.
+        Objects which are unreachable from ECMAScript may be released after long periods
+        of time, or never at all. For this reason, this specification uses the term
+        "may" when describing behavior triggered by garbage collection.
+      </p>
+
+      <p>
+        The semantics of WeakRefs and FinalizationGroups is based on two
+        operations which happen at particular points in time:
+      </p>
+
+      <ul>
+        <li>
+          When WeakRef.prototype.deref is called, the referent (if it's not
+          already dead) is kept alive so that subsequent, synchronous accesses also
+          return the object. This list is reset when synchronous work is done using
+          the ClearKeptObjects abstract operation.
+        </li>
+
+        <li>
+          When an object which is registered with a FinalizationGroup becomes
+          unreachable, a call of the FinalizationGroup's cleanup callback may
+          eventually be made, after synchronous JavaScript execution completes.
+          The FinalizationGroup cleanup is performed with the
+          FinalizationGroupCleanup abstract operation.
```

Comment:
Looks like the operation is named `CleanupFinalizationGroup` in the rest of the document.

Reply:
Ah, right. Interested in making a PR to correct it?